include/dfu/dfu_target.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef DFU_TARGET_H__
+#define DFU_TARGET_H__
+
+/**
+ * @defgroup dfu_target DFU Target
+ * @brief DFU Target
+ *
+ * @{
+ */
+
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Functions which needs to be supported by all DFU targets.
+ */
+struct dfu_target {
+	int (*init)(void);
+	int (*offset_get)(size_t *offset);
+	int (*write)(const void *const buf, size_t len);
+	int (*done)(void);
+};
+
+/**
+ * @brief Find the image type for the buffer of bytes recived. Used to determine
+ *	  what dfu target to initialize.
+ *
+ * @param buf[in] A buffer of bytes which are the start of an binary firmware image.
+ * @param len[in] The length of the provided buffer.
+ *
+ * @return Positive identifier for a supported image type or a negative error
+ *	   code identicating reason of failure.
+ **/
+int dfu_target_img_type(const void *const buf, size_t len);
+
+/**
+ * @brief Initialize the resources needed for the specific image type DFU
+ *	  target.
```

Comment:
Mention that the offset must always be checked after calling `_init()`.

Reply:
Do you mean that it is not valid to call it before `_init()` or that it is required that you call `_offset_get` when you have called `_init()` ?

Reply:
The last. The fact that the offset can be non-zero is important to know.